core/src/main/java/tachyon/worker/hierarchy/StorageDir.java

```diff
@@ -469,6 +469,17 @@ public TachyonURI getDirPath() {
   }
 
   /**
+   * Get last access time of a block in current StorageDir
+   *
+   * @return the last block access time in Ms, -1 if the block doesn't exist
+   */
+  public long getLastBlockAccessTimeMs(long blockId) {
+    Long lastBlockAccessTimeMs = mLastBlockAccessTimeMs.get(blockId);
+    return lastBlockAccessTimeMs != null ? lastBlockAccessTimeMs : -1;
```

Comment:
Maybe the non-existent time should be a constant, and not just '-1'. It would be easier to do comparisons, if needed.